vmm/src/lib.rs

```diff
@@ -2444,6 +2450,7 @@ mod tests {
             cpu_template: Some(CpuFeaturesTemplate::T2),
         };
         assert!(vmm.set_vm_configuration(machine_config).is_err());
+        assert_eq!(vmm.vm_config.mem_size_mib, Some(128));
```

Comment:
The mem_size_mib is 128 not because it was reset to default but because its value hasn't change between the previous call to to `set_vm_configuration` and this one. I suggest removing this check.